src/Psalm/Internal/LanguageServer/LanguageServer.php

```diff
@@ -352,14 +373,34 @@ function (IssueData $issue_data) use ($file_path) : Diagnostic {
                             $diagnostic_severity = DiagnosticSeverity::ERROR;
                             break;
                     }
-                    // TODO: copy issue code in 'json' format
-                    return new Diagnostic(
+                    $diagnostic = new Diagnostic(
                         $description,
                         $range,
                         null,
                         $diagnostic_severity,
                         'Psalm'
                     );
+
+                    $code = 'PS' . \str_pad((string) $issue_data->shortcode, 3, "0", \STR_PAD_LEFT);
+
+                    if ($this->project_analyzer->language_server_use_extended_diagnostic_codes) {
+                        // Added in VSCode 1.43.0 and will be part of the LSP 3.16.0 standard.
+                        // Since this new functionality is not backwards compatible, we use a
+                        // configuration option so the end user must opt in to it using the cli argument.
+                        // https://github.com/microsoft/vscode/blob/1.43.0/src/vs/vscode.d.ts#L4688-L4699
+
+                        /** @psalm-suppress InvalidPropertyAssignmentValue */
+                        $diagnostic->code = [
+                            "value" => $code,
+                            "target" => $issue_data->link,
+                        ];
```

Comment:
here we send the detailed diagnostic code with the help link.

Reply:
@muglug here is where it uses the short code to sent to the client.  Right now it pads it to 3 chars and prepends 'PS' to it so a code of 22 would be 'PS022'.  We can make it be anything you like, even making the code sent to the client be the link also; we would just set the link to both "value" and "target".  We need to send the extended code structure instead of just a string in order to make it clickable in the IDE.